homeassistant/components/reolink/light.py

```diff
@@ -0,0 +1,153 @@
+
+    supported: Callable[[Host, int | None], bool] = lambda api, ch: True
+    get_brightness: Callable[[Host, int | None], int] | None = None
+    set_brightness: Callable[[Host, int | None, float], Any] | None = None
+
+
+LIGHT_ENTITIES = (
+    ReolinkLightEntityDescription(
+        key="floodlight",
+        name="Floodlight",
+        icon="mdi:spotlight-beam",
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        is_on=lambda api, ch: api.whiteled_state(ch),
+        turn_on_off=lambda api, ch, value: api.set_whiteled(ch, state=value),
+        get_brightness=lambda api, ch: api.whiteled_brightness(ch),
+        set_brightness=lambda api, ch, value: api.set_whiteled(
+            ch, brightness=int(value)
+        ),
+    ),
+    ReolinkLightEntityDescription(
+        key="ir_lights",
+        name="Infra red lights in night mode",
+        icon="mdi:led-off",
+        supported=lambda api, ch: api.supported(ch, "ir_lights"),
+        is_on=lambda api, ch: api.ir_enabled(ch),
+        turn_on_off=lambda api, ch, value: api.set_ir_lights(ch, value),
+    ),
+    ReolinkLightEntityDescription(
+        key="status_led",
+        name="Status LED",
+        icon="mdi:lightning-bolt-circle",
+        supported=lambda api, ch: api.supported(ch, "status_led"),
+        is_on=lambda api, ch: api.status_led_enabled(ch),
+        turn_on_off=lambda api, ch, value: api.set_status_led(ch, value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink light entities."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ReolinkLightEntity(reolink_data, channel, entity_description)
+        for entity_description in LIGHT_ENTITIES
+        for channel in reolink_data.host.api.channels
+        if entity_description.supported(reolink_data.host.api, channel)
+    )
+
+
+class ReolinkLightEntity(ReolinkCoordinatorEntity, LightEntity):
+    """Base light entity class for Reolink IP cameras."""
+
+    entity_description: ReolinkLightEntityDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
+        entity_description: ReolinkLightEntityDescription,
+    ) -> None:
+        """Initialize Reolink light entity."""
+        super().__init__(reolink_data, channel)
+        self.entity_description = entity_description
+
+        self._attr_unique_id = (
+            f"{self._host.unique_id}_{self._channel}_{entity_description.key}"
+        )
+
+        if self.entity_description.set_brightness is None:
+            self._attr_supported_color_modes = {ColorMode.ONOFF}
+            self._attr_color_mode = ColorMode.ONOFF
+        else:
+            self._attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+            self._attr_color_mode = ColorMode.BRIGHTNESS
```

Comment:
This could be defined in the entity_description

Reply:
It is not part of the default LightEntityDescription, so I would still need these lines to set the attributes, and need to add both supported_color_modes and color_mode to the enitity description of each light, so that will be more lines of code.

Currently it is more compact, I like it better this way.

But if you insist I can change it to be in the entity_description.

Reply:
No, I expected those attributes to be exposed from the description but if that is not expected by the light platform then its nothing I would request being changed.